fedimintd/src/fedimintd.rs

```diff
@@ -94,6 +95,24 @@ pub struct ServerOpts {
     #[arg(long, env = FM_EXTRA_DKG_META_ENV, value_parser = parse_map, default_value="")]
     extra_dkg_meta: BTreeMap<String, String>,
 
+    /// Default Api secret that will be used to communicate with other peers
+    /// secret
+    ///
+    /// If Federation is public, it should have `api_secret` and
+    /// `api_extra_secrets` not set.
+    ///
+    /// Otherwise it will enforce authentication on the Api HTTP level.
+    #[arg(env = FM_API_SECRET_ENV)]
+    api_secret: Option<String>,
+
+    /// Comma separated list of additional secrets that can be used to access
+    /// the Api
+    ///
+    /// Format: `secret1,secret2,...`
+    #[arg(env = FM_API_EXTRA_SECRETS_ENV, value_parser = parse_vec, default_value="")]
+    // Note: full path used due to https://github.com/clap-rs/clap/issues/4626#issuecomment-1593681582
+    api_extra_secrets: std::vec::Vec<String>,
```

Comment:
Why make this two options :sob:

Reply:
The peer uses an Api to talk with other peers. So there's a difference between "one I accept and use" and "one I just still accept". @elsirion

Reply:
So would you want to communicate a warning to the client if they connect via an extra secret? Otherwise how does behavior differs between the two? If behavior doesn't differ, why not just leave it to the operators to put the active one first so they remember which ones to phase out?

Reply:
> If behavior doesn't differ, why not just leave it to the operators to put the active one first so they remember which ones to phase out?

It could be done, but I thought having distinct fields will be self-describing and bring the meaning and function to the user.


Possibly the naming could be improved. Should it be `FM_ACTIVE_API_SECRET` and `FM_STALE_API_SECRETS`?

> So would you want to communicate a warning to the client if they connect via an extra secret?

I thought about it, but I don't see any easy way to do it, so I left it out for now. We could possibly add another API call that end user app could periodically make to verify and warn the user?

@elsirion